codex/raw/marketplace.md

```diff
@@ -0,0 +1,275 @@
+---
+title: CODEX-MARKETPLACE
+name: Codex Storage Marketplace
+status: raw
+tags: codex
+editor: Dmitriy <dryajov@status.im>
+contributors:
+- Mark <mark@codex.storage>
+- Adam <adam.u@status.im>
+- Eric <ericmastro@status.im>
+- Jimmy Debe <jimmy@status.im>
+---
+
+## Abstract
+
+This specification describes a method for Codex storage providers and client nodes to participate in a storage marketplace. 
+The goal is to create a storage marketplace that promotes durability.
+
+## Motivation
+The Codex network aims to create a peer-to-peer storage engine with strong data durability, 
+data persistence guarantees and node storage incentives.
+Support for resource restricted devices, like mobile devices should also be embraced.
+The protocol should remove complexity to allow for a simple implementation and 
+simplify incentive mechanisms.
+
+## Semantics 
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### Definitions
+
+| Terminology  | Description |
+| --------------- | --------- |
+| storage providers | A Codex node that provides storage services to the marketplace. |
+| validator nodes | A Codex node that checks for missing storage proofs and triggers for a reward. |
+| client nodes | The most common Codex node that interacts with other nodes to store, locate and retrieve data. |
+| slots | Created by client nodes when a new dataset is requested to be stored. Discussed further in the [slots section](#slots).  |
+
+### Storage Request
+
+Client nodes can create storage requests on the Codex network via the Codex marketplace.
+The marketplace is a smart contract deployed on a EVM-compatiable blockchain.
+The smart contract MUST handle storage request, storage slot state, 
+storage provider rewards, storage provider collateral, and storage proof state. 
+
+To create a request to store a dataset on the Codex network,
+client nodes MUST split the dataset into data chunks, $(c_1, c_2, c_3, \ldots, c_{n})$.
+Using an erasure coding technique, 
+the data chunks are encoded and placed into separate slots.
+The erasure coding technique SHOULD be the [Reed-Soloman algorithm](https://hackmd.io/FB58eZQoTNm-dnhu0Y1XnA).
+
+The requester, client node, SHOULD submit a transaction with the desired request parameters.
```

Comment:
```suggestion
When the client node is prompted by the user to create a storage request, it MUST submit a transaction with the desired request parameters.
```

Submitting a transaction is a MUST, otherwise the Storage Request won't ever get to the Storage Providers and the Storage Request will never be filled.